internal/scorer/algorithm/registry_test.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2022 Criticality Score Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//	https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package algorithm
+
+import (
+	"reflect"
+	"testing"
+)
+
+type testAlgo struct{}
+
+func (t testAlgo) Score(record map[string]float64) float64 {
+	return 0
+}
+
+func TestNewAlgorithm(t *testing.T) {
+	type args struct {
```

Comment:
A better approach to this test method might be to:

1. populate the registry with one or two entries before the test loop (I suggest having a factory that always works and stores `inputs`, and a factory that always returns an error).
2. define tests that call `NewAlgorithm` on the registry.

i.e:

```go
func TestNewAlgorithm(t *testing.T) {
  r := NewRegistry()
  r.Register("test", func(...) { ... })

  tests := []struct{
    // ...
  }{
    // ...
  }
  for _, test := range tests {
    t.Run(...)
  }
```

Reply:
Using a custom registry is a good idea as it avoids side-effect and removes the need to do cleanups.

Reply:
Sorry, I don't understand what you mean by "custom registry", could you please explain and give some examples? It's been a while since I worked on this PR :)

Reply:
When I say custom registry I mean the one returned by `r := NewRegistry()`, rather than using the `GlobalRegistry`.

By creating a new `Registry` in each test function, you avoid the need to run a cleanup.

Reply:
I can't use `NewRegistry()` because `Register()` and `NewAlgorithm()` are based on `GlobalRegistry` meaning that they won't run properly without setting `GlobalRegistry`. 

For example, if we used `r := NewRegistry()` and then registered `r` using `r.Register()`. `GlobalRegistry` would be empty when calling `NewAlgorithm()`.